proposals/esm-integration/README.md

```diff
@@ -88,57 +88,74 @@ Three things happen for each module during the Construction phase.
+For JavaScript ES modules, an export name is bound to a slot on the Module Record. For all of the values that we currently allow, this slot will contain a reference to another memory location, which contains the object (e.g. the `WebAssembly.Global` object that is being imported).
+
+The term "snapshot" means that WebAssembly will snapshot the reference. This does result in an observable difference between the way JavaScript and WebAssembly handle updates. In JavaScript modules, it is possible for the exporting module to update an export to point to a different object. Other JavaScript modules that are importing that export will then get the new object. In contrast, while WebAssembly modules will see changes to the object that the export was first bound to (e.g. when the exporting module calls `WebAssembly.Global.prototype.set` to change the value), the WebAssembly module will never see updated bindings.
 
 ## Examples and explanation of import/export handling
 
 Some imports are challenging to handle in the current WebAssembly specification. In this section, we will discuss each kind of export and what the specification described above means for how it is handled.
 
 ### wasm imports <- JS exports
 
-| export type | value (not a WebAssembly.Global)* | global | memory | table | function |
-|-------------|------------------|--------|--------|-------|----------|
-|              | Error                         | Error  | Error  | Error | snapshot |
+When the Wasm module is evaluated, the exported value of the JS module is used in a way which is based on the import type declared in the WebAssembly module.
 
-\*While WebAssembly only has the concept of globals, JS could export either a regular JS value or a `WebAssembly.Global`.
+| import type | behavior |
+|-------------|----------|
+| global      | If the exported value is a WebAssembly.Global object, throw an exception if types mismatch, otherwise take that object as the import. Otherwise, if the imported type is `const`, cast the imported value to the appropriate numeric type, and create a new const global to hold the result. Otherwise, throw an exception. |
+| memory      | Check that the exported value is a WebAssembly.Memory object which meets the type imported; use it if so, otherwise, throw an exception |
+| table       | Check that the exported value is a WebAssembly.Table object which meets the type imported; use it if so, otherwise, throw an exception |
+| function    | If the exported value is a WebAssembly exported function based on the same types, make use of it. Otherwise, [create a host function](https://webassembly.github.io/spec/js-api/index.html#create-a-host-function) out of the JS function which includes the casts implied by its type. |
+
+While WebAssembly only has the concept of globals, JS could export either a regular JS value or a `WebAssembly.Global`.
 
 The sequence of operations for a wasm module which depends on a JS module is as follows:
 
 1. wasm module is parsed
```

Comment:
nit: parsed -> decoded

Reply:
I could also say "compiled" here... not sure what is clearest. Parse is what the Wasm core specification uses, but probably it's not clearest for most developers.

Reply:
The core spec says "decode" for binary and "parse" for text format. Maybe "decode and compile"? The compilation step isn't actually observable (which is why the core spec does not mention it), but given that the API exposes these as `compile` functions it makes sense to make the intention explicit. Alternatively, there could be a note pointing out the intention.

Reply:
Thanks, I'll say "compile" then.